src/LayerPlanBuffer.cpp

```diff
@@ -107,6 +110,17 @@ void LayerPlanBuffer::addConnectingTravelMove(LayerPlan* prev_layer, const Layer
             // we force the path to use retraction
             path.retract = true;
         }
+        if (path.retract && !path.unretract_before_last_travel_move &&
+            next_extruder_settings.get<bool>("outer_inset_first") &&
+            (next_extruder_settings.get<coord_t>("infill_line_distance") == 0 ||
+                    !next_extruder_settings.get<bool>("infill_before_walls")))
+        {
+            // If there is a retraction and the next layer's first path is an outer wall, then we need to make sure that
```

Comment:
What happens if the next object has per-object settings that change these setting values? I think it's getting them from the wrong settings object.

Getting the first line is a complex business. Other things that could be printed in between are:
* Support
* Draft shield
* Ooze shield
* Prime tower
* Meshes with per-object settings causing the outer wall to not be the first thing printed.

For the Flow Advance algorithm I wrote a 160-line function to get the flow rate of the first extrusion on the next layer. Complex business.